Domains/4-Application/Connectors/GeologicalModel/GeologicalModel.ecschema.xml

```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ECSchema schemaName="GeologicalModel" alias="geomodel" version="01.00.00" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2" displayLabel="Geological Model" description="The GeologicalModel schema defines classes that represent data for interpreting the earth's subterranean structure">
+    <ECSchemaReference name="BisCore" version="01.00.14" alias="bis" />
+    <ECSchemaReference name="CoreCustomAttributes" version="01.00.03" alias="CoreCA"/>
+    <ECSchemaReference name="Analytical" version="01.00.00" alias="anlyt" />
+
+    <ECEntityClass typeName="GeologicAnalyticalPartition" modifier="Sealed" description="A specialized Analytical Partition advertising a Geologic model (interpretation) based on Fault Blocks partitioned by one or more Fault Surfaces.">
+        <BaseClass>anlyt:AnalyticalPartition</BaseClass>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="GeologicalModel" modifier="Sealed" description="An AnalyticalModel specialization expected to contain GeologicElement3d instances at different granularities.">
+        <BaseClass>anlyt:AnalyticalModel</BaseClass>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="GeologicElement3d" modifier="Abstract" description="An AnalyticalElement serving as common base class for all element-classes modeling Geological concepts.">
+        <BaseClass>anlyt:AnalyticalElement</BaseClass>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="FaultBlock" modifier="None" description="A GeologicElement3d modeling a set of volumes and surfaces that may be partitioned by one or more Fault Surfaces">
+        <BaseClass>GeologicElement3d</BaseClass>
```

Comment:
What's the best way to associate `GeologicalCategoryVolume` and `ContactSurface` instances to `FaultBlock` instances? Volumes and Contact Surfaces "belong" to a particular Fault Block.

Reply:
That's what the sentence in the remarks address:

"_Instances of `GeologicalCategoryVolume` are expected to be contained in a submodel of a `FaultBlock` instance that it is part of._"

What that means is that whoever creates a `FaultBlock` instance (e.g. a Connector) needs to create a submodel for it. Then, instances of `GeologicalCategoryVolume` or `ContactSurface` associated to such `FaultBlock` needs to be created in the appropriate submodel.

For further illustration, consider the example at: https://www.itwinjs.org/learning/backend/createelements/#geometricelement3d ... which inserts an element in the iModel.

The equivalent code for a `GeologicalCategoryVolume` instance would look like:

```
const props = {
    model: faultBlockSubmodelId, // submodel created earlier, when the correspoinding FaultBlock instance was inserted into the iModel
    classFullName: "geologicalModel:GeologicalCategoryVolume",
    ...
  };
  return iModelDb.elements.insertElement(props);
```
In summary, BIS currently doesn't have a schema-based technique to enforce the kind of model expected for a particular element-class. Currently, the only way to add constrains/validation in such situation is via a "domain-handler" - Typescript code written for a particular schema, reacting to certain operations (e.g. OnInsert) so it has the opportunity to check any needed domain-specific conditions and fail the operation if they are not met. However, writing a "domain-handler" for a schema is not mandatory, so it is generally the data-writer's responsibility (i.e. your connector) to follow the expected element-organization documented in a schema (if any).

Reply:
Ah, that explains it. Thanks, @diegoalexdiaz . The team is looking at using PCF for this connector. @jchick-bentley , is this technique supported in PCF?

Reply:
@danywu I'm not aware implementations that use the ModelModelsElement pattern proposed here, but, at the same time I don't see any obvious limitations.  When I last discussed PCF, I said if you can picture your data model in tabular format like an excel file.  It is difficult to picture how this pattern could be represented in a single spread sheet, but, it the xlsx loader could be extended to accommodate multiple sheets and it is not the only loader option.  It's easier to see this pattern represented in JSON format and there's a JSON loader.  I'm optimistic we can get it to work.  I recommend scheduling this task ahead of others to identify any impediments early.